diff -ruN cryptography-45.0.7/Cargo.toml cryptography-45.0.7-subint/Cargo.toml
--- cryptography-45.0.7/Cargo.toml	2025-09-01 07:05:40.000000000 -0400
+++ cryptography-45.0.7-subint/Cargo.toml	2025-09-15 18:19:01.779000000 -0400
@@ -22,8 +22,8 @@
 
 [workspace.dependencies]
 asn1 = { version = "0.21.3", default-features = false }
-pyo3 = { version = "0.25", features = ["abi3"] }
-pyo3-build-config = { version = "0.25" }
+pyo3 = { version = "0.26", features = ["abi3", "unsafe-allow-subinterpreters" ], path="/tmp/SBo/pyo3-0.26.0-subint" }
+pyo3-build-config = { version = "0.26", path="/tmp/SBo/pyo3-0.26.0-subint/pyo3-build-config" }
 openssl = "0.10.72"
 openssl-sys = "0.9.108"
 
